06/ObjectVectorExample.cpp

```diff
@@ -0,0 +1,43 @@
+#include <algorithm>
+#include <iostream>
+#include <vector>
+#include "ObjectVectorExample.h"
+#include "Score.h"
+
+using namespace std;
+
+namespace samples
+{
+	void ObjectVectorExample()
+	{
+		vector<Score> myScoreList;
+		myScoreList.reserve(5);
+
+		myScoreList.push_back(Score(30, "class 1"));
+		myScoreList.push_back(Score(59, "class 2"));
+		myScoreList.push_back(Score(87, "Introducing C++"));
+		myScoreList.push_back(Score(74, "class 3"));
+		myScoreList.push_back(Score(41, "class 4"));
+
+		cout << "Before erasing the element : ";
+		for (vector<Score>::iterator iter = myScoreList.begin(); iter != myScoreList.end(); iter++)
+		{
+			(*iter).PrintVariables();
+		}
+		cout << endl;
+
+		cout << "After erasing the element : ";
+		for (vector<Score>::iterator iter = myScoreList.begin(); iter != myScoreList.end();)
+		{
+			if ((*iter).GetClassName() == "Introducing C++")
+			{
+				iter = myScoreList.erase(iter);
+			}
+			else
+			{
+				(*iter).PrintVariables();
+				iter++;
+			}
+		}
+	}
+}
```

Comment:
maybe we need another example here... i wanna show this

1. insert score objects into vector
2. find an object from iterator and change the value.
3. show it doesn't actually change from the list.

I'm not sure if number 2 has to be

iter->SetScore(10);

or

Score score = *iter;
score.SetScore(xx);

if it's former that's better, if former doesn't work use latter. 

i want to show them it's object that's stored in vector<> not *.. so changing the value directly won't have any impact